components/canvas/webgl_thread.rs

```diff
@@ -667,8 +667,13 @@ impl WebGLThread {
     fn handle_swap_buffers(
         &mut self,
         swap_ids: Vec<SwapChainId>,
-        completed_sender: WebGLSender<()>,
+        completed_sender: WebGLSender<u64>,
+        _sent_time: u64,
     ) {
+        #[cfg(feature = "xr-profile")]
+        let start_swap = time::precise_time_ns();
+        #[cfg(feature = "xr-profile")]
+        println!("WEBXR PROFILING [swap request]:\t{}ms", (start_swap - _sent_time) as f64 / 1_000_000.);
```

Comment:
Let's define another `to_ms` function in canvas_traits to avoid repeating this calculation.